ssh-server/main.go

```diff
@@ -66,11 +71,56 @@ func getConfiguration() (*ssh.ServerConfig, error) {
 		return nil, fmt.Errorf("creating signer: %v", err)
 	}
 
-	config := &ssh.ServerConfig{NoClientAuth: true}
+	config := &ssh.ServerConfig{
+		PublicKeyCallback: func(conn ssh.ConnMetadata, key ssh.PublicKey) (*ssh.Permissions, error) {
+			authorizedKeys, err := getAuthorizedKeys()
+			if err != nil {
+				return nil, fmt.Errorf("getting the authorized keys: %v", err)
+			}
+
+			if !authorizedKeys[string(key.Marshal())] {
+				return nil, fmt.Errorf("authorizing key: %v", err)
```

Comment:
There is no error here; you are reusing a value from somewhere else
You need to mint an error with errors.New

Reply:
done